src/main/java/net/library/service/UserService.java

```diff
@@ -33,14 +36,18 @@ public class UserService {
 
     private final UserRepository userRepository;
 
+    public User getUserByUsername(String username) {
+        return userRepository.findByUsername(username).orElseThrow(() -> new UsernameNotFoundException("User not found"));
+    }
+
     public List<User> getAllUsers() {
         return userRepository.findAll().stream().toList();
     }
 
     public Page<UserDto> getAllByFilter(String startDate, String endDate, String userName, String moderationState, String userState, String roleType, Pageable pageable) {
         log.info("Filtering with username: {}, startDate: {}, endDate: {}", userName, startDate, endDate);
 
-        UserValidationService.isLength(userName, 3);
+        PhoneValidationService.isLength(userName, 3);
```

Comment:
deleteAll -  Add @Transactional annotation to ensure atomicity (consistent with deleteById at line 84) and add logging before the operation
But of course it is better to not use this method at all